src/sugarblas.jl

```diff
@@ -18,24 +18,22 @@ function blascopy!(n::Number, DX::AbstractArray, incx::Integer, DY::AbstractArra
 end
 
 # Add robustness to scal!
-function scal!(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
+@inbounds function scal!(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
     i = 1
     n = min(n,length(DX))
-    while i <= n
+    @simd for i in 1:incx:n
         DX[i] *= DA
-        i += incx
     end
     DX
 end
 
 # Add robustness to scal
-function scal{T}(n::Integer, DA::Number, DX::AbstractArray{T}, incx::Integer)
+@inbounds function scal{T}(n::Integer, DA::Number, DX::AbstractArray{T}, incx::Integer)
     zeros(T, size(DX)...)
     i = 1
     n = min(n,length(DX))
-    while i <= n
+    @simd for i in 1:incx:n
```

Comment:
I don't think simd works when you are skipping over unknown number of elements. You could check if `incx = 1` and have a special simd loop for `1:n`